var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -221,6 +221,7 @@ class Llvm(CMakePackage, CudaPackage):
         description="Enable code-signing on macOS",
     )
     variant("python", default=False, description="Install python bindings")
+    variant("lua", default=True, when="@11: +lldb", description="Enable lua scripting inside lldb")
```

Comment:
Maybe should be LLVM 10, but from 11 they fix the lua version to 5.3, so `@11:` simplifies the package.py a bit.

Reply:
Yeah, it is needed for LLVM 10 but I didn't dig into it much further.

Reply:
@haampie are you aware of any changes in Spack that changed the treatment of the conditional variants? Will `spack install llvm@~lldb~lua` or `spack install llvm@:10~lua` work in this case? If not, I would like this to be replaced with a `conflict` like it's done for other variants around.

Reply:
`spack install llvm@:10~lua` won't work, but why is that problematic? 🤔

Reply:
Because when I say `llvm~lua` (e.g. in the configuration files) I mean "I don't want lua" and not "I want llvm version 11 or newer".